results/cmd/api/main_test.go

```diff
@@ -186,8 +432,92 @@ func TestDeleteTaskRun(t *testing.T) {
 		t.Fatalf("failed to delete taskrun: %v", r.String())
 	}
 
-	// Check if a deleted taskrun can be delete again
+	// Check if a deleted taskrun can be deleted again
 	if _, err := srv.DeleteTaskRunResult(ctx, &pb.DeleteTaskRunRequest{ResultsId: r.GetResultsId()}); status.Code(err) != codes.NotFound {
 		t.Fatalf("same taskrun not supposed to be deleted again: %v", r.String())
 	}
 }
+func TestListTaskRuns(t *testing.T) {
+	// Create a temporary database
+	srv, err := setupTestDB(t)
+	if err != nil {
+		t.Fatalf("failed to setup db: %v", err)
+	}
+
+	// Connect to fake server and create a new taskrun
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	// Create a bunch of taskruns for testing
+	t1 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "31415926",
+			Name:      "taskrun",
+			Namespace: "default",
+		},
+	}
+	t2 := &pb.TaskRun{
+		ApiVersion: "v1alpha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "43245243",
+			Name:      "task",
+			Namespace: "default",
+		},
+	}
+	t3 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "1234556",
+			Name:      "mytaskrun",
+			Namespace: "demo",
+		},
+	}
+	t4 := &pb.TaskRun{
+		ApiVersion: "v1beta1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "demo",
+		},
+	}
+	t5 := &pb.TaskRun{
+		ApiVersion: "v1apha1",
+		Metadata: &pb.ObjectMeta{
+			Uid:       "543535",
+			Name:      "newtaskrun",
+			Namespace: "demo",
+		},
+	}
+	testList := []*pb.TaskRun{t1, t2, t3, t4, t5}
+	expect := []string{t1.String(), t3.String(), t4.String()}
+	sort.Strings(expect)
```

Comment:
It's a good practice to group variables together where they are used. Since this isn't used until the very last comparison, we should move these down there.

Reply:
done